payments/paypal/__init__.py

```diff
@@ -310,15 +310,19 @@ def release(self, payment):
         self.post(payment, url)
 
     def refund(self, payment, amount=None):
-        if amount is None:
-            amount = payment.captured_amount
-        amount_data = self.get_amount_data(payment, amount)
-        refund_data = {"amount": amount_data}
+        refund_data = {}
+        if amount is not None:
+            refund_data["amount"] = self.get_amount_data(payment, amount)
```

Comment:
I'm not sure that I follow; why does using `payment.captured_amount` not work here?

Reply:
@WhyNotHugo, in the ideal world it would. However, we do not live in one. E.g. since 2021, every (our) PayPal payment model has the captured amount equal to zero due to https://github.com/jazzband/django-payments/issues/309. (And such asynchrony between reality and models can easily happen due to other bugs as well.)